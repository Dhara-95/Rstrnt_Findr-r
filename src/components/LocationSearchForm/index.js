import React from "react";

const LocationSearchForm = () => {
  return (
    <div className="py-8 flex justify-around items-center w-1/2 mx-auto">
      <span className="text-grey">Find restaurants near you:</span>
      <div className="rounded bg-white w-1/2 flex py-1 pr-1 pl-5 justify-between">
        <input
          className="w-3/5 focus:outline-none"
          name="address"
          type="text"
          placeholder="W13 3RH"
        />
        <button
          className="submit-button rounded py-2 px-8 bg-teal hover:bg-teal-hover active:bg-teal-click focus:outline-none"
          type="button"
          value="Find"
        >
          Find
        </button>
      </div>
    </div>
  );
};

export default LocationSearchForm;
