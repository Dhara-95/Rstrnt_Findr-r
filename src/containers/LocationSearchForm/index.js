import React from "react";
import { useHistory } from "react-router-dom";

import SubmitButton from "../../components/SubmitButton";
import { LabelsObject } from "../../Labels";

const LocationSearchForm = (props) => {
  const { setUserInput } = props;
  const {
    findRestaurantsLabel,
    findButtonLabel,
    searchResultURLLabel,
  } = LabelsObject;

  const history = useHistory();

  return (
    <div className="py-8 flex flex-col lg:justify-around lg:flex-row xl:justify-around xl:flex-row 2xl:justify-around 2xl:flex-row items-center w-1/2 mx-auto">
      <span className="text-grey">{findRestaurantsLabel}</span>
      <div className="rounded bg-white lg:w-1/2 xl:w-1/2 2xl:w-1/2 flex py-1 pr-1 pl-5 justify-between">
        <input
          onChange={(e) => setUserInput(e.target.value)}
          className="w-3/5 focus:outline-none"
          name="postcode"
          type="text"
          placeholder="W13 3RH"
        />
        <SubmitButton
          label={findButtonLabel}
          history={history}
          route={searchResultURLLabel}
        />
      </div>
    </div>
  );
};

export default LocationSearchForm;
