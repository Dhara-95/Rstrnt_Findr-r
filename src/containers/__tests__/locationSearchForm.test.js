import React from "react";
import renderer from "react-test-renderer";

import { LocationSearchForm } from "../LocationSearchForm";

it("displays the Search Form with the correct components", () => {
  const tree = renderer
    .create(
      <LocationSearchForm
        setUserInput={"function"}
        handleButtonClick={"function"}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
