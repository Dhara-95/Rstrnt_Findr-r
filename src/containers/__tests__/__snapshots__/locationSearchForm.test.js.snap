// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`displays the Search Form with the correct components 1`] = `
<div
  className="py-8 flex flex-col lg:justify-around lg:flex-row xl:justify-around xl:flex-row 2xl:justify-around 2xl:flex-row items-center w-1/2 mx-auto"
>
  <span
    className="text-grey"
  >
    Find restaurants near you:
  </span>
  <div
    className="rounded bg-white lg:w-1/2 xl:w-1/2 2xl:w-1/2 flex py-1 pr-1 pl-5 justify-between"
  >
    <input
      className="w-3/5 focus:outline-none"
      name="postcode"
      onChange={[Function]}
      placeholder="W13 3RH"
      type="text"
    />
    <button
      className="submit-button rounded py-2 px-8 bg-teal hover:bg-teal-hover active:bg-teal-click focus:outline-none"
      onClick={[Function]}
      type="button"
      value="Find"
    >
      Find
    </button>
  </div>
</div>
`;
